data/frame_type.go

```diff
@@ -44,10 +44,26 @@ const (
 	// Soon?
 	// "timeseries-wide-ohlc" -- known fields for open/high/low/close
 	// "histogram" -- BucketMin, BucketMax, values...
-	// "directory-listing" -- known fields for name, size, mime-type, modified, etc
 	// "trace" -- ??
 	// "node-graph-nodes"
 	// "node-graph-edges"
+
+	// FrameTypeDirectoryListing represents the items in a directory
+	// field[0]:
+	//  * name
+	//  * new paths can be constructed from the parent path + separator + name
+	// field[1]:
+	//  * media-type
+	//  * when "directory" it can be nested
+	FrameTypeDirectoryListing = "directory-listing"
+
+	// FrameTypeTable represents an arbitrary table structure with no constraints
+	FrameTypeTable = "table"
+
+	// FrameTypeTableFooter may exist next to FrameTypeTable and represent data that
+	// should show up in the footer section.  It must have the same width, but not
+	// necessarily same field types as the sibling table data
+	FrameTypeTableFooter = "table-footer"
```

Comment:
@scottlepp -- can we use this to indicate the footer vs body?

Reply:
works for me.  do we want something explicit on the field to map fields from frames together (besides name).  the field names in salesforce between frames are not exactly the same.  doing this currently
```
			field.Config.Custom = map[string]interface{}{
				"referenceField": "some-field-from-detail-frame",
				"reducer":        "min, max, etc",
			}
```

Reply:
Lets evaluate that independently -- the field types/meta is an different issue from frame types.  In the example above, I don't know why they would be mixed